knack/config.py

```diff
@@ -109,9 +109,9 @@ def sections(self):
 
     def items(self, section):
         import re
-        pattern = self.env_var_name(section, '.+')
+        pattern = self.env_var_name(section, '[0-9A-Z]+')
```

Comment:
The option name can consist of multiple parts, like `AZURE_BATCHAI_STORAGE_ACCOUNT`.